File: spring-data-requery/src/test/java/org/springframework/data/requery/domain/sample/AbstractOrder.java
package org.springframework.data.requery.domain.sample;

import io.requery.*;
import org.springframework.data.requery.domain.AbstractPersistable;

/**
 * AbstractOrder
 *
 * @author debop@coupang.com
 * @since 18. 6. 25
 */
@Entity
@Table(name = "ORDERS")
public abstract class AbstractOrder extends AbstractPersistable<Long> {

    @Key
    @Generated
    protected Long id;

    @ForeignKey
    @ManyToOne
    protected Customer customer;

    protected String orderNo;


    private static final long serialVersionUID = 3606365166891885684L;
}

File: spring-data-requery/src/test/kotlin/org/springframework/data/requery/domain/sample/AbstractOrder.kt
package org.springframework.data.requery.domain.sample

import io.requery.Entity
import io.requery.ForeignKey
import io.requery.Key
import io.requery.ManyToOne
import org.springframework.data.requery.domain.AbstractPersistable
import java.util.*

@Entity
abstract class AbstractOrder: AbstractPersistable<Long>() {

    @get:Key
    abstract override val id: Long

    @get:ForeignKey
    @get:ManyToOne
    abstract var customer: AbstractCustomer?

    abstract var orderNo: String

    override fun hashCode(): Int = Objects.hashCode(orderNo)

}